### PR TITLE
Clear previous entries before submitting

### PR DESCRIPTION
**Pull request by lamchuntat**

### CODE REVIEW

**Comment by lamchuntat** [comment]
Currently, if dry_run is set to True, `submit_clockify` does nothing

**Comment by lamchuntat** [comment]
@ThorpeJosh @siewyeng @changukshin 

Needs review

**Review by siewyeng — APPROVED** [review]
LGTM

**Comment by lamchuntat** [comment]
Requires a working API key for unit test to work. Will wait for https://github.com/ThorpeJosh/tp-timesheet/pull/75 to be merged to adopt the pytest fixture.

File: tp_timesheet/tests/test_remove_existing_entries.py [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+
+
+def test_remove_existing_entries():
+    """Test for resubmission of clockify"""
+
+    # Instantiate a Clockify object
+    clockify = Clockify("your_api_here")
+
+    # Add a clockify entry for 1 January 2022
+    clockify.submit_clockify(datetime.date(2022, 12, 5), 1)
```
Comment (ThorpeJosh) [inline tp_timesheet/tests/test_remove_existing_entries.py:14]: I think this is Dec 5th not Jan 1st

File: tp_timesheet/tests/test_remove_existing_entries.py [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+
+    # Resubmit clockify entry for 1 January 2022
+    clockify.submit_clockify(datetime.date(2022, 12, 5), 4)
+
+    # Check only one entry exists for 1 January 2022
+    number_of_entries = len(clockify._get_time_entry_id(datetime.date(2022, 12, 5)))
+    assert (number_of_entries == 100), f"Incorrect number of entries, expected: 100, returned: {number_of_entries}"
```
Comment (ThorpeJosh) [inline tp_timesheet/tests/test_remove_existing_entries.py:21]: Why 100?

File: tp_timesheet/clockify_timesheet.py [hunk 3]
```diff
@@ -68,6 +79,39 @@ def submit_clockify(self, date, working_hours, dry_run=False):
             )
             response.raise_for_status()
 
+    def _get_time_entry_id(self, date):
+        """Get a time entry from clockify on a certain date"""
+        time_entry_id = []
```
Comment (ThorpeJosh) [inline tp_timesheet/clockify_timesheet.py:84]: I'd rename this `time_entry_ids` to help imply it is a list. Usually lists are plural

File: tp_timesheet/tests/test_remove_existing_entries.py [hunk 4]
```diff
@@ -0,0 +1,24 @@
+"""Unit test for removing same-day existing entries before submitting a new clockify entry"""
+
+import datetime
+from tp_timesheet.clockify_timesheet import Clockify
+
+
+def test_remove_existing_entries():
+    """Test for resubmission of clockify"""
```
Comment (ThorpeJosh) [inline tp_timesheet/tests/test_remove_existing_entries.py:8]: The test title and docstring seem different. The test is also submitting twice and then testing only 1 exists.
I'd probably test at each step so none exists initially, then 1, then 1 again, then zero, etc

File: tp_timesheet/tests/test_remove_existing_entries.py [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+    clockify.submit_clockify(datetime.date(2022, 12, 5), 1)
+
+    # Resubmit clockify entry for 1 January 2022
+    clockify.submit_clockify(datetime.date(2022, 12, 5), 4)
+
+    # Check only one entry exists for 1 January 2022
+    number_of_entries = len(clockify._get_time_entry_id(datetime.date(2022, 12, 5)))
+    assert (number_of_entries == 100), f"Incorrect number of entries, expected: 100, returned: {number_of_entries}"
+
+    # Remove all entries for 1 January 2022
```
Comment (ThorpeJosh) [inline tp_timesheet/tests/test_remove_existing_entries.py:23]: We want to make sure this runs regardless of a failure or not. Not sure of best way to do this though. Fixtures can help, should look up pytest docs.

**Review by siewyeng — APPROVED** [review]

**Pull request merged**